src/map/mob.cpp

```diff
@@ -1263,6 +1263,24 @@ static int32 mob_can_changetarget(struct mob_data* md, struct block_list* target
 	}
 }
 
+/**
+ * Randomizes the target ID of a monster if it has the given mode
+ * @param bl Unit that is going to attack
+ * @param target_id Target ID to modify
+ */
+void mob_randomtarget(mob_data& md, int32& target_id) {
```

Comment:
Why do you pass the target_id as a reference instead of just passing the value?

Reply:
So that we can update it. Saves you having to reassign the value and also saves memory.